### PR TITLE
Implement contiguous access to acc_block in get_matrix_elements

### PR DESCRIPTION
**Pull request by davidbowler**
Change GEMM call in get_matrix_elements (transpose) so that access to acc_block is contiguous within innermost loop, and replace inner loop with array segment.

### CODE REVIEW

**Review by lionelalexandre — APPROVED** [review]

**Review by tkoskela — APPROVED** [review]
Profiling on Vtune showed 6x better performance in the inner loop. I trust you've checked the results are still correct. Of course you can wait until #184 is merged and the CI will tell you :smile:

**Comment by davidbowler** [comment]
Results appear to be fine, so I'm merging.

**Pull request merged**
